src/include/btree_cmp_inline.h

```diff
@@ -166,6 +306,145 @@ __wt_compare_bounds(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_ITEM *key, u
     return (0);
 }
 
+#ifdef HAVE_X86INTRIN_H
+/*
+ * __lex_compare_skip_gt_16 --
+ *     Lexicographic comparison routine for data greater than 16 bytes, skipping leading bytes.
+ *     Returns: < 0 if user_item is lexicographically < tree_item = 0 if user_item is
+ *     lexicographically = tree_item > 0 if user_item is lexicographically > tree_item We use the
+ *     names "user" and "tree" so it's clear in the btree code which the application is looking at
+ *     when we call its comparison function.
+ */
+static WT_INLINE int
+__lex_compare_skip_gt_16(
+  const uint8_t *ustartp, const uint8_t *tstartp, size_t len, int lencmp, size_t *matchp)
+{
+    __m128i res_eq, t, u;
+    uint64_t t64, u64;
+    int32_t eq_bits;
+    size_t match, final_match;
+    const uint8_t *tendp, *treep, *uendp, *userp;
+
+    match = *matchp;
+    uendp = ustartp + len;
+    tendp = tstartp + len;
+
+    /*
+     * Compare 16 bytes at a time until we find a difference or run out of 16 byte chunks to
+     * compare.
+     */
+    for (userp = ustartp + match, treep = tstartp + match; uendp - userp > WT_VECTOR_SIZE;
+         userp += WT_VECTOR_SIZE, treep += WT_VECTOR_SIZE) {
+        u = _mm_loadu_si128((const __m128i *)userp);
+        t = _mm_loadu_si128((const __m128i *)treep);
+        res_eq = _mm_cmpeq_epi8(u, t);
+        if ((eq_bits = _mm_movemask_epi8(res_eq)) != 65535) {
+            match = (size_t)(userp - ustartp);
+            goto final128;
+        }
+    }
+
+    /*
+     * Rewind until there is exactly 16 bytes left. We know we started with at least 16, so we are
+     * still in bound.
+     */
+    u = _mm_loadu_si128((const __m128i *)(uendp - WT_VECTOR_SIZE));
+    t = _mm_loadu_si128((const __m128i *)(tendp - WT_VECTOR_SIZE));
+    res_eq = _mm_cmpeq_epi8(u, t);
+    eq_bits = _mm_movemask_epi8(res_eq);
+
+final128:
```

Comment:
I think you want this to be in the `else` block because we only jump here when we find a difference.

Reply:
yes, true.